src/main/java/sg/dex/starfish/impl/memory/AMemoryOperation.java

```diff
@@ -19,7 +23,33 @@
 public abstract class AMemoryOperation extends AMemoryAsset implements Operation {
 
     protected AMemoryOperation(String metaString, MemoryAgent memoryAgent) {
-        super(metaString, memoryAgent);
+        super(getMetaWithProv(metaString,memoryAgent), memoryAgent);
```

Comment:
Don't create provenance when a metadata String is provided directly. Only do it when default metadata is being created. 

Please write tests to verify this.